lib/MahoCLI/Commands/SysEncryptionKeyRegenerate.php

```diff
@@ -0,0 +1,244 @@
+        ]);
+
+        if (\Composer\InstalledVersions::isInstalled('phpseclib/mcrypt_compat')) {
+            $output->writeln('');
+            $output->writeln('<error>Warning: phpseclib/mcrypt_compat is installed. This package can cause encryption issues and should be removed.</error>');
+            $output->writeln('<error>Please remove it using: composer remove phpseclib/mcrypt_compat</error>');
+            $output->writeln('');
+        }
+
+        return Command::SUCCESS;
+    }
+
+    private function decrypt(#[\SensitiveParameter] string $data): string
+    {
+        if ($this->isOldEncryptionKeyM1 && function_exists('mcrypt_module_open')) {
+            $key = $this->oldEncryptionKey;
+            $handler = mcrypt_module_open(MCRYPT_BLOWFISH, '', MCRYPT_MODE_ECB, ''); // @phpstan-ignore constant.notFound,constant.notFound
+            $initVector = mcrypt_create_iv(mcrypt_enc_get_iv_size($handler), MCRYPT_RAND); // @phpstan-ignore function.notFound,function.notFound,constant.notFound
+            mcrypt_generic_init($handler, $key, $initVector); // @phpstan-ignore function.notFound
+            $data = mdecrypt_generic($handler, (string) base64_decode((string) $data)); // @phpstan-ignore function.notFound
+            mcrypt_generic_deinit($handler); // @phpstan-ignore function.notFound
+            mcrypt_module_close($handler); // @phpstan-ignore function.notFound
+            return str_replace("\x0", '', trim($data));
+        }
+
+        try {
+            $decoded = sodium_base642bin($data, SODIUM_BASE64_VARIANT_ORIGINAL);
+        } catch (\SodiumException $e) {
+            $exception = new \Exception('Invalid base64 encoding: ' . $e->getMessage());
+            Mage::logException($exception);
+            return '';
+        }
+
+        $key = sodium_hex2bin($this->oldEncryptionKey);
+        $nonce = mb_substr($decoded, 0, SODIUM_CRYPTO_SECRETBOX_NONCEBYTES, '8bit');
+        $ciphertext = mb_substr($decoded, SODIUM_CRYPTO_SECRETBOX_NONCEBYTES, null, '8bit');
+        $plaintext = sodium_crypto_secretbox_open($ciphertext, $nonce, $key);
+
+        // Clean sensitive data from memory
+        sodium_memzero($data);
+        sodium_memzero($decoded);
+        sodium_memzero($key);
+        sodium_memzero($nonce);
+        sodium_memzero($ciphertext);
+
+        return (string) $plaintext;
+    }
+
+    private function encrypt(#[\SensitiveParameter] string $data): string
+    {
+        $key = sodium_hex2bin($this->newEncryptionKey);
+        $nonce = random_bytes(SODIUM_CRYPTO_SECRETBOX_NONCEBYTES);
+        $ciphertext = sodium_crypto_secretbox($data, $nonce, $key);
+        $encrypted = sodium_bin2base64($nonce . $ciphertext, SODIUM_BASE64_VARIANT_ORIGINAL);
+
+        // Clean sensitive data from memory
+        sodium_memzero($data);
+        sodium_memzero($key);
+        sodium_memzero($nonce);
+        sodium_memzero($ciphertext);
+
+        return $encrypted;
+    }
+}
```

Comment:
```suggestion
    private function createBackupPath(): string
    {
        $i = 0;
        $currentDate = date('Y-m-d');

        do {
            $backupPath = 'app/etc/local.xml.bak.' . $currentDate;
            if ($i > 0) {
                $backupPath .= "-$i";
            }
            $i++;
        } while (file_exists($backupPath));

        return $backupPath;
    }
}
```

Do you think we should have a function like this? What happened is that I ran `./maho sys:encryptionkey:regenerate` twice (intentionally) but that meant that I lost my original encryption key.

Reply:
mmm instead of that I'll add HHMMSS but I thought it would never happen in production because there's 1 or 2 confirmations asked before actually doing anything

Reply:
what do you think about the latest version?

Reply:
Yeah, that works too and is much simpler. I think better safe than sorry, because you never know what a user might do.

Reply:
:-) ;-)